src/android/registerNativeModule.js

```diff
@@ -1,36 +1,46 @@
-const readFile = require('./fs').readFile;
-const writeFile = require('./fs').writeFile;
-const compose = require('lodash').flowRight;
+const fs = require('fs');
 const getReactVersion = require('../getReactNativeVersion');
 const getPrefix = require('./getPrefix');
+const isInstalled = require('./isInstalled');
 
-const applyPatch = (filePath, patch) =>
-  compose(writeFile(filePath), patch, readFile(filePath));
+const applyPatch = require('./patches/applyPatch');
+const makeStringsPatch = require('./patches/makeStringsPatch');
+const makeSettingsPatch = require(`./patches/makeSettingsPatch`);
+const makeBuildPatch = require(`./patches/makeBuildPatch`);
 
-module.exports = function registerNativeAndroidModule(name, androidConfig, params, projectConfig) {
+module.exports = function registerNativeAndroidModule(
+  name,
+  androidConfig,
+  params,
+  projectConfig
+) {
+
+  if (isInstalled(projectConfig, name)) {
```

Comment:
I am saying that this check is done inside `link.js` on this line (https://github.com/rnpm/rnpm-plugin-link/blob/master/src/link.js#L35) and your implementation should follow what's currently in master (no checks, no return value https://github.com/rnpm/rnpm-plugin-link/blob/master/src/android/registerNativeModule.js#L9-L36).